cmd/fleet/userAgent.go

```diff
@@ -66,8 +66,8 @@ func validateUserAgent(r *http.Request, verConst version.Constraints) error {
 	if !strings.HasPrefix(userAgent, userAgentPrefix) {
 		return ErrInvalidUserAgent
 	}
-	verStr := strings.TrimSpace(strings.TrimSuffix(strings.TrimPrefix(userAgent, userAgentPrefix), "-snapshot"))
-	ver, err := version.NewVersion(verStr)
+	verSep := strings.Split(strings.TrimPrefix(userAgent, userAgentPrefix), "-")
```

Comment:
Is ok for now.  Prefer to see this logic centralized and better commented.
@aleksmaus  has a PR that conflicts so can be fixed up then.